### PR TITLE
Add 338 release notes

### PR DESCRIPTION
**Pull request by martint**
Closes https://github.com/prestosql/presto/issues/4246

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-338.rst [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+General Changes
+---------------
+
+* Fix incorrect results when joining tables on a masked column. (:issue:`4251`)
+* Fix planning failure when multiple columns have a mask. (:issue:`4322`)
+* Fix incorrect comparison for ``TIMESTAMP WITH TIME ZONE`` values with precision larger than 3. (:issue:`4305`)
+* Fix query failure when using ``VALUES`` with a ``real`` or ``double`` ``NaN`` value. (:issue:`4119`)
+* Fix query failure when joining tables on a ``real`` or ``double`` column and one of the joined tables
+  contains ``NaN`` value. (:issue:`4272`)
+* Fix incorrect query results when Parquet file has no min/max statistics for an integer column. (:issue:`4200`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-338.rst:14]: is "integer" the right term to describe a smallint/integer/bigint types?

File: presto-docs/src/main/sphinx/release/release-338.rst [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+Hive Connector Changes
+----------------------
+
+* Fix query failure when reading from a table partitioned on a ``real`` or ``double`` column containing
+  a ``NaN`` value. (:issue:`4266`)
+* Do not require caching directories to be configured on coordinator. (:issue:`3987`, :issue:`4280`)
+* Fix Azure ADL caching support. (:issue:`4240`)
+* Add support for setting column comments via the ``COMMENT ON COLUMN`` syntax. (:issue:`2516`)
+* Add hidden ``$partition`` column to partitioned table. (:issue:`3582`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-338.rst:38]: "Expose"?

File: presto-docs/src/main/sphinx/release/release-338.rst (hunk 1 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-338.rst:15]: Maybe "integral" would be more appropriate.

File: presto-docs/src/main/sphinx/release/release-338.rst [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+=========================
+
+General Changes
+---------------
+
+* Fix incorrect results when joining tables on a masked column. (:issue:`4251`)
+* Fix planning failure when multiple columns have a mask. (:issue:`4322`)
+* Fix incorrect comparison for ``TIMESTAMP WITH TIME ZONE`` values with precision larger than 3. (:issue:`4305`)
+* Fix incorrect rounding for timestamps before 1970-01-01. (:issue:`4370`)
+* Fix query failure when using ``VALUES`` with a ``real`` or ``double`` ``NaN`` value. (:issue:`4119`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-338.rst:12]: How about
```
with a floating point ``NaN`` value

File: presto-docs/src/main/sphinx/release/release-338.rst [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+* Fix query failure when joining tables on a ``real`` or ``double`` column and one of the joined tables
+  contains ``NaN`` value. (:issue:`4272`)
+* Fix incorrect query results when Parquet file has no min/max statistics for an integral column. (:issue:`4200`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-338.rst:15]: This is a Hive change

File: presto-docs/src/main/sphinx/release/release-338.rst [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+* Fix unauthorized error for internal requests to management endpoints. (:issue:`4304`)
+* Fix memory leak while using dynamic filtering. (:issue:`4228`)
+* Improve dynamic partition pruning for broadcast joins. (:issue:`4262`)
+* Add support for setting column comments via the ``COMMENT ON COLUMN`` syntax. (:issue:`2516`)
+* Add compatibility mode for non-compliant clients when rendering datetime type names with default precision
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-338.rst:20]: How about "legacy" here rather than non-compliant

File: presto-docs/src/main/sphinx/release/release-338.rst [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  in ``information_schema`` tables. This can be enabled via the ``deprecated.omit-datetime-type-precision``
+  configuration property or ``omit_datetime_type_precision`` session property. (:issue:`4349`, :issue:`4377`)
+* Enforce ``NOT NULL`` column declarations when writing data. (:issue:`4144`)
+
+JDBC Driver Changes
+-------------------
+
+* Fix excessive CPU usage when reading query results. (:issue:`3928`)
+* Implement DatabaseMetaData#getClientInfoProperties. (:issue:`4318`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-338.rst:29]:
```
Implement ``DatabaseMetaData.getClientInfoProperties()``.

File: presto-docs/src/main/sphinx/release/release-338.rst [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+* Add support for reading numeric values encoded as strings. (:issue:`4341`)
+
+Hive Connector Changes
+----------------------
+
+* Fix query failure when reading from a table partitioned on a ``real`` or ``double`` column containing
+  a ``NaN`` value. (:issue:`4266`)
+* Fix sporadic failure when writing to bucketed sorted tables on S3. (:issue:`2296`)
+* Fix handling of strings when translating Hive views. (:issue:`3266`)
+* Do not require caching directories to be configured on coordinator. (:issue:`3987`, :issue:`4280`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-338.rst:43]: caching -> cache

File: presto-docs/src/main/sphinx/release/release-338.rst [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+* Fix Azure ADL caching support. (:issue:`4240`)
+* Add support for setting column comments via the ``COMMENT ON COLUMN`` syntax. (:issue:`2516`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-338.rst:45]: We can shorten this to

> Add support for setting column comments.

Since the main item contains the syntax. And in the future releases, we would not mention the syntax when a connector adds support.

File: presto-docs/src/main/sphinx/release/release-338.rst [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+* Expose hidden ``$partition`` column to partitioned tables. (:issue:`3582`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-338.rst:46]:
```
Add hidden ``$partition`` column for partitioned tables that contains the partition name.

File: presto-docs/src/main/sphinx/release/release-338.rst [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+Kafka Connector Changes
+-----------------------
+
+* Fix query failure when a column is projected and also referenced in a query predicate
+  when reading from Kafka topic using ``RAW`` decoder. (:issue:`4183`)
+
+MySQL Connector Changes
+-----------------------
+
+* Fix type mapping for unsigned integer types in MySQL connector. (:issue:`4187`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-338.rst:57]: > Fix type mapping for unsigned integer types.

File: presto-docs/src/main/sphinx/release/release-338.rst [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+
+Oracle Connector Changes
+------------------------
+
+* Filter internal objects (e.g. ``sys``) when listing schemas. (:issue:`3784`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-338.rst:62]: Exclude internal schemas (e.g., ``sys``) from schema listings.

File: presto-docs/src/main/sphinx/release/release-338.rst [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+* Add support for connection pooling. (:issue:`3770`)
+
+Base-JDBC Connector Library Changes
+-----------------------------------
+
+* Filter objects on hidden schemas when listing tables. (:issue:`3834`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-338.rst:68]:
```
Exclude the underlying database's ``information_schema`` from schema listings.

File: presto-docs/src/main/sphinx/release/release-338.rst (hunk 5 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-338.rst:20]: I wanted to make emphasis on the fact that they shouldn't have hardcoded those assumptions, but I'll update as suggested.
